### ISSUE TITLE
V9 Enterprise Install

### ISSUE THREAD

**Issue by bwadz** [odoo/odoo#8999]
I have decided to load the V9 Enterprise addons. I pulled the enterprise addons from github and started the odoo server with the enterprise addons.  --ERROR

I edited the configuration file to put the enterprise addons before everything else - ERROR

I was told to copy the web addon from the enterprise addons folder to the standard folder - ERROR
odoo@cloud-server-01:~/odoo$ ./odoo.py
2015-10-10 00:28:33,172 20633 INFO ? openerp: OpenERP version 9.0
2015-10-10 00:28:33,173 20633 INFO ? openerp: addons paths: ['/home/odoo/.local/share/Odoo/addons/9.0', u'/home/odoo/enterprise', u'/home/odoo/odoo/addons', u'/home/odoo/odoo/openerp/addons', u'/home/odoo/odoo']
2015-10-10 00:28:33,173 20633 INFO ? openerp: database: default@default:default
2015-10-10 00:28:33,491 20633 INFO ? openerp.service.server: HTTP service (werkzeug) running on 0.0.0.0:8069
2015-10-10 00:28:40,742 20633 WARNING ? openerp.addons.website_crm_score.models.sales_team: flanker not found, email validation disabled.
2015-10-10 00:28:40,943 20633 INFO ? openerp.addons.bus.models.bus: Bus.loop listen imbus on db postgres
2015-10-10 00:28:41,645 20633 INFO ? openerp.addons.report.models.report: You need Wkhtmltopdf to print a pdf version of the reports.
2015-10-10 00:28:41,934 20633 INFO ? openerp.http: HTTP Configuring static files
2015-10-10 00:28:41,944 20633 INFO v9test openerp.modules.loading: loading 1 modules...
2015-10-10 00:28:41,957 20633 INFO v9test openerp.modules.loading: 1 modules loaded in 0.01s, 0 queries
2015-10-10 00:28:41,976 20633 INFO v9test openerp.modules.loading: loading 35 modules...
2015-10-10 00:28:42,046 20633 INFO v9test openerp.modules.loading: 35 modules loaded in 0.07s, 0 queries
2015-10-10 00:28:42,542 20633 INFO v9test openerp.modules.loading: Modules loaded.
2015-10-10 00:28:42,543 20633 INFO v9test openerp.addons.base.ir.ir_http: Generating routing map
2015-10-10 00:28:42,736 20633 INFO v9test werkzeug: 98.167.207.193 - - [10/Oct/2015 00:28:42] "GET / HTTP/1.1" 200 -
2015-10-10 00:28:43,147 20633 INFO v9test werkzeug: 98.167.207.193 - - [10/Oct/2015 00:28:43] "GET /web HTTP/1.1" 500 -
2015-10-10 00:28:43,174 20633 ERROR v9test werkzeug: Error on request:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/werkzeug/serving.py", line 177, in run_wsgi
    execute(self.server.app)
  File "/usr/local/lib/python2.7/dist-packages/werkzeug/serving.py", line 165, in execute
    application_iter = app(environ, start_response)
  File "/home/odoo/odoo/openerp/service/server.py", line 245, in app
    return self.app(e, s)
  File "/home/odoo/odoo/openerp/service/wsgi_server.py", line 184, in application
    return application_unproxied(environ, start_response)
  File "/home/odoo/odoo/openerp/service/wsgi_server.py", line 170, in application_unproxied
    result = handler(environ, start_response)
  File "/home/odoo/odoo/openerp/http.py", line 1416, in __call__
    return self.dispatch(environ, start_response)
  File "/home/odoo/odoo/openerp/http.py", line 1390, in __call__
    return self.app(environ, start_wrapped)
  File "/usr/local/lib/python2.7/dist-packages/werkzeug/wsgi.py", line 588, in __call__
    return self.app(environ, start_response)
  File "/home/odoo/odoo/openerp/http.py", line 1580, in dispatch
    result = ir_http._dispatch()
  File "/home/odoo/odoo/addons/web_editor/models/ir_http.py", line 16, in _dispatch
    return super(ir_http, self)._dispatch()
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_http.py", line 176, in _dispatch
    return self._handle_exception(e)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_http.py", line 147, in _handle_exception
    return request._handle_exception(exception)
  File "/home/odoo/odoo/openerp/http.py", line 737, in _handle_exception
    return super(HttpRequest, self)._handle_exception(exception)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_http.py", line 172, in _dispatch
    result = request.dispatch()
  File "/home/odoo/odoo/openerp/http.py", line 768, in dispatch
    r = self._call_function(**self.params)
  File "/home/odoo/odoo/openerp/http.py", line 315, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/odoo/odoo/openerp/service/model.py", line 113, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/odoo/odoo/openerp/http.py", line 311, in checked_call
    result.flatten()
  File "/home/odoo/odoo/openerp/http.py", line 1367, in flatten
    self.response.append(self.render())
  File "/home/odoo/odoo/openerp/http.py", line 1360, in render
    context=request.context)
  File "/home/odoo/odoo/openerp/api.py", line 223, in wrapper
    return old_api(self, *args, **kwargs)
  File "/home/odoo/odoo/addons/web_editor/models/ir_ui_view.py", line 29, in render
    return super(view, self).render(cr, uid, id_or_xml_id, values=values, engine=engine, context=context)
  File "/home/odoo/odoo/openerp/api.py", line 223, in wrapper
    return old_api(self, *args, **kwargs)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_ui_view.py", line 1065, in render
    return self.pool[engine].render(cr, uid, id_or_xml_id, qcontext, loader=loader, context=context)
  File "/home/odoo/odoo/openerp/api.py", line 223, in wrapper
    return old_api(self, *args, **kwargs)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 255, in render
    return self.render_node(element, qwebcontext, generated_attributes=qwebcontext.pop('generated_attributes', ''))
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 298, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 321, in render_element
    generated_attributes= name == "t" and generated_attributes or ''))
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 296, in render_node
    result = self._render_tag[t_render](self, element, template_attributes, generated_attributes, qwebcontext)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 449, in render_tag_call
    d[0] = self.render_element(element, template_attributes, generated_attributes, d)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 321, in render_element
    generated_attributes= name == "t" and generated_attributes or ''))
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 298, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 321, in render_element
    generated_attributes= name == "t" and generated_attributes or ''))
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 298, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 321, in render_element
    generated_attributes= name == "t" and generated_attributes or ''))
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 296, in render_node
    result = self._render_tag[t_render](self, element, template_attributes, generated_attributes, qwebcontext)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 460, in render_tag_call
    res = self.render(cr, uid, template, d)
  File "/home/odoo/odoo/openerp/api.py", line 223, in wrapper
    return old_api(self, *args, **kwargs)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 255, in render
    return self.render_node(element, qwebcontext, generated_attributes=qwebcontext.pop('generated_attributes', ''))
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 298, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 321, in render_element
    generated_attributes= name == "t" and generated_attributes or ''))
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 298, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 321, in render_element
    generated_attributes= name == "t" and generated_attributes or ''))
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 296, in render_node
    result = self._render_tag[t_render](self, element, template_attributes, generated_attributes, qwebcontext)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 389, in render_tag_foreach
    enum = self.eval_object(expr, qwebcontext)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 199, in eval_object
    return self.eval(expr, qwebcontext)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 196, in eval
    raise_qweb_exception(message="Could not evaluate expression %r" % expr, expression=expr, template=template)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 193, in eval
    return qwebcontext.safe_eval(expr)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 106, in safe_eval
    return eval(expr, None, locals_dict, nocopy=True, locals_builtins=True)
  File "/home/odoo/odoo/openerp/tools/safe_eval.py", line 298, in safe_eval
    return eval(c, globals_dict, locals_dict)
  File "", line 1, in <module>
    
QWebException: "'NoneType' object has no attribute '__getitem__'" while evaluating
"menu_data['children']"

**Comment by wtaferner** [comment]
@bwadz 
It is unfortunately because you do need to remove the community web module manually as it is loaded first and does not work with the rest of enterprise stuff. 

It would be wise to provide some kind of automatic detection and load the correct web module. I tried to play around a little bit but it seems to break a lot if you try something like auto detection and only load the correct module. One way or the other...

@antonylesuisse 
Do you think someone within Odoo could implement such an auto detection b/c it is really annoying that you need these extra steps while developing and switching between these two versions where the main difference in core is the duplicated web module.
Could you explain why you took this approach which destroys modularity?

**Comment by antonylesuisse** [comment]
Use --addons-path=odoo-enterprise,odoo-community/addons

**Comment by wtaferner** [comment]
Ok, I tried that before but it was not working, but it seems that this issue was fixed lately...thank you!

**Comment by bwadz** [comment]
I must not be as smart as you guys :)

I downloaded the enterprise branch - loaded it in the $Home directory.  Home now contains odoo, enterprise.  I then tried starting the server with the enterprise addons folder first in the addons-path.  I verified that the enterprise path was first in the server log.
![ss- 2015-10-12 at 7 45 37 am](https://cloud.githubusercontent.com/assets/9170865/10430669/4b489b14-70b5-11e5-89c4-dc8b5007f1b9.png)
The server sits and poll normally until I connect via a browser. --ERROR
2015-10-12 14:47:24,849 2448 ERROR v9test werkzeug: Error on request:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/werkzeug/serving.py", line 177, in run_wsgi
    execute(self.server.app)
  File "/usr/local/lib/python2.7/dist-packages/werkzeug/serving.py", line 165, in execute
    application_iter = app(environ, start_response)
  File "/home/odoo/odoo/openerp/service/server.py", line 245, in app
    return self.app(e, s)
  File "/home/odoo/odoo/openerp/service/wsgi_server.py", line 184, in application
    return application_unproxied(environ, start_response)
  File "/home/odoo/odoo/openerp/service/wsgi_server.py", line 170, in application_unproxied
    result = handler(environ, start_response)
  File "/home/odoo/odoo/openerp/http.py", line 1416, in __call__
    return self.dispatch(environ, start_response)
  File "/home/odoo/odoo/openerp/http.py", line 1390, in __call__
    return self.app(environ, start_wrapped)
  File "/usr/local/lib/python2.7/dist-packages/werkzeug/wsgi.py", line 588, in __call__
    return self.app(environ, start_response)
  File "/home/odoo/odoo/openerp/http.py", line 1580, in dispatch
    result = ir_http._dispatch()
  File "/home/odoo/odoo/addons/utm/models/ir_http.py", line 13, in _dispatch
    response = super(ir_http, self)._dispatch()
  File "/home/odoo/odoo/addons/website/models/ir_http.py", line 190, in _dispatch
    resp = super(ir_http, self)._dispatch()
  File "/home/odoo/odoo/addons/web_editor/models/ir_http.py", line 16, in _dispatch
    return super(ir_http, self)._dispatch()
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_http.py", line 176, in _dispatch
    return self._handle_exception(e)
  File "/home/odoo/odoo/addons/website/models/ir_http.py", line 239, in _handle_exception
    return super(ir_http, self)._handle_exception(exception)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_http.py", line 147, in _handle_exception
    return request._handle_exception(exception)
  File "/home/odoo/odoo/openerp/http.py", line 737, in _handle_exception
    return super(HttpRequest, self)._handle_exception(exception)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_http.py", line 172, in _dispatch
    result = request.dispatch()
  File "/home/odoo/odoo/openerp/http.py", line 768, in dispatch
    r = self._call_function(**self.params)
  File "/home/odoo/odoo/openerp/http.py", line 315, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/odoo/odoo/openerp/service/model.py", line 113, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/odoo/odoo/openerp/http.py", line 311, in checked_call
    result.flatten()
  File "/home/odoo/odoo/openerp/http.py", line 1367, in flatten
    self.response.append(self.render())
  File "/home/odoo/odoo/openerp/http.py", line 1360, in render
    context=request.context)
  File "/home/odoo/odoo/openerp/api.py", line 223, in wrapper
    return old_api(self, *args, **kwargs)
  File "/home/odoo/odoo/addons/website/models/ir_ui_view.py", line 143, in render
    return super(view, self).render(cr, uid, id_or_xml_id, values=values, engine=engine, context=context)
  File "/home/odoo/odoo/openerp/api.py", line 223, in wrapper
    return old_api(self, *args, **kwargs)
  File "/home/odoo/odoo/addons/web_editor/models/ir_ui_view.py", line 29, in render
    return super(view, self).render(cr, uid, id_or_xml_id, values=values, engine=engine, context=context)
  File "/home/odoo/odoo/openerp/api.py", line 223, in wrapper
    return old_api(self, *args, **kwargs)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_ui_view.py", line 1065, in render
    return self.pool[engine].render(cr, uid, id_or_xml_id, qcontext, loader=loader, context=context)
  File "/home/odoo/odoo/openerp/api.py", line 223, in wrapper
    return old_api(self, *args, **kwargs)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 255, in render
    return self.render_node(element, qwebcontext, generated_attributes=qwebcontext.pop('generated_attributes', ''))
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 298, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 321, in render_element
    generated_attributes= name == "t" and generated_attributes or ''))
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 296, in render_node
    result = self._render_tag[t_render](self, element, template_attributes, generated_attributes, qwebcontext)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 449, in render_tag_call
    d[0] = self.render_element(element, template_attributes, generated_attributes, d)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 321, in render_element
    generated_attributes= name == "t" and generated_attributes or ''))
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 298, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 321, in render_element
    generated_attributes= name == "t" and generated_attributes or ''))
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 298, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 321, in render_element
    generated_attributes= name == "t" and generated_attributes or ''))
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 296, in render_node
    result = self._render_tag[t_render](self, element, template_attributes, generated_attributes, qwebcontext)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 460, in render_tag_call
    res = self.render(cr, uid, template, d)
  File "/home/odoo/odoo/openerp/api.py", line 223, in wrapper
    return old_api(self, *args, **kwargs)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 255, in render
    return self.render_node(element, qwebcontext, generated_attributes=qwebcontext.pop('generated_attributes', ''))
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 298, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 321, in render_element
    generated_attributes= name == "t" and generated_attributes or ''))
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 298, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 321, in render_element
    generated_attributes= name == "t" and generated_attributes or ''))
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 296, in render_node
    result = self._render_tag[t_render](self, element, template_attributes, generated_attributes, qwebcontext)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 389, in render_tag_foreach
    enum = self.eval_object(expr, qwebcontext)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 199, in eval_object
    return self.eval(expr, qwebcontext)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 196, in eval
    raise_qweb_exception(message="Could not evaluate expression %r" % expr, expression=expr, template=template)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 193, in eval
    return qwebcontext.safe_eval(expr)
  File "/home/odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 106, in safe_eval
    return eval(expr, None, locals_dict, nocopy=True, locals_builtins=True)
  File "/home/odoo/odoo/openerp/tools/safe_eval.py", line 298, in safe_eval
    return eval(c, globals_dict, locals_dict)
  File "", line 1, in <module>
    
QWebException: "'NoneType' object has no attribute '__getitem__'" while evaluating
"menu_data['children']"
2015-10-12 14:47:44,155 2448 INFO v9test werkzeug: 98.167.207.193 - - [12/Oct/2015 14:47:44] "POST /calendar/notify HTTP/1.1" 200 -

*********************

**Comment by bwadz** [comment]
Ok, I finally figured out my issue with the install of V9 Enterprise. I am sure this is common knowledge to everyone else, but just in case... I am going to document my issue in hopes that it might help someone in the future.

The issue...
I installed Odoo V9 community edition and created a database to play around with the setup and look at some potential development.  I later wanted to investigate the V9 Enterprise version and downloaded the enterprise addons and started the server with the addons starting with enterprise and then the standaard odoo addons.  

I was getting the error above and it was recommended that I manually remove the odoo/addons/web addon and allow the enterprise version to load in its place.

The serve would not serve the Odoo backend only the web frontend.

Solution---
I only had a single database created that was initialized with the community edition. Once I loaded the enterprise version, went to manage database and created one with the enterprise version, it initialized and loaded without problems. 

Unfortunately this little issue took my 3 days to figure out - Slow and steady and as a result a little bit smarter :)

Good luck
